### ISSUE TITLE
[FEATURE] Allow user to set resource constraint for Longhorn components

### ISSUE THREAD

**Issue by PhanLe1010** [longhorn/longhorn#8332]
## Is your feature request related to a problem? Please describe (👍 if you like this request)

In the current implementation of Longhorn, there is no setting to control the CPU and memory limits as we believe that Longhorn, as a critical storage layer, should not be limited or killed by OOMKill. The reason is that if Longhorn struggles, the application workloads cannot function well. However, this assumption might not be correct for the edge environment in which the user might want to set limits for the Longhorn system toa void Longhorn reader the system unusable. 

There are multiple tickets related to the resource constraint topic currently:
1. Share manager: https://github.com/longhorn/longhorn/issues/8314
1. Instance manager: https://github.com/longhorn/longhorn/issues/4009
1. And I imagine that user would likely want to customized the setting for other components like longhorn-manager, csi-plugin,...

This ticket server as the master ticket to combine them all together. 

## Additional Context

@innobead mentioned that Rook resource constraints strategy is: https://github.com/rook/rook/blob/master/design/ceph/resource-constraints.md . Maybe we can take a look at their approach

**Comment by benfiola** [comment]
Apologies in advance, but could this be expanded to support - broadly - 'pod metadata' (using terminology [from the Alertmanager/Prometheus operators](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md))?  For my personal use case, I'd like to be able to attach additional labels to Longhorn managed pods (in addition to constraining pod requests).

Also, I'm more than happy to implement this if you think it's an appropriate first-issue for a newcomer like myself and if there's an appetite for external contributions.

**Comment by fmunteanu** [comment]
Why such an important feature keeps getting postponed? Devs are now targeting 1.9.0 release.

**Comment by AyushChothe** [comment]
Hi @derekbit and @ejweber, Please implement this feature as soon as possible. It’s a crucial feature for any company, and it’s been a significant blocker for us for a long time.

**Comment by mossroy** [comment]
Resource constraints are not only about resource limits, it's also about resource requests.
K8s uses resource requests to schedule pods on nodes.
If you don't set memory requests for example, k8s might decide to schedule some pods on nodes that don't have enough memory, and cause nodes to get out of memory. That's what happens to me.

To fix that issue, I believe we would just need to be able to set theses resource requests/limits in values.yaml (as users of the Helm chart), like it was started in https://github.com/longhorn/longhorn/pull/8307
No need to provide default values, that can be difficult to decide, as they really depend on how longhorn is really used.
What is missing is to let users of the Helm chart set their own resource requests/limits

**Comment by fmunteanu** [comment]
Still waiting on this important feature to be implemented.
